src/datastructure/GlobalDescriptor.cpp

```diff
@@ -78,6 +86,23 @@ GlobalDescriptorDataType GlobalDescriptor::getDataType() const
     return m_dataType;
 }
 
+bool GlobalDescriptor::isValid() 
```

Comment:
Is this needed, if the builder is used and the checks are already done? The goal is that you cannot create an ill-formed object.